aero.minova.core.application.system.service/src/main/java/aero/minova/core/application/system/controller/XSqlProcedureController.java

```diff
@@ -178,4 +200,112 @@ XSqlProcedureResult findxSqlResultSet(String idToFind, List<XSqlProcedureResult>
+		sqlRequest.addColumn(new Column("KeyText", DataType.STRING));
+		sqlRequest.addColumn(new Column("TransactionChecker", DataType.STRING));
+
+		List<Row> inputRows = new ArrayList<>();
+		for (XTable xTable : inputTables) {
+			Row requestParam = new Row();
+			sqlRequest.getRows().add(requestParam);
+			requestParam.addValue(new Value(xTable.getTable().getName(), null));
+			requestParam.addValue(null);
+
+			inputRows.add(requestParam);
+		}
+		try {
+			Table viewResult = securityService.getTableForSecurityCheck(sqlRequest);
+
+			// Wir müssen ja eigentlich einen Eintrag in der Datenbank dazu haben, sonst hätten wir sie bisher nicht ausführen können.
+			if (viewResult.getRows().size() == 0) {
+				throw new RuntimeException("msg.PrivilegeError");
+			}
+
+			// Im Prinzip den ganzen Spaß nochmal nur mit den TransactionChecker.
+			List<XTable> xtables = new ArrayList<>();
+			for (Row row : viewResult.getRows()) {
+				if (row.getValues().size() != 0 && row.getValues().get(1) != null) {
+					String dependencyTableName = row.getValues().get(0).getStringValue();
+					String transactionChecker = row.getValues().get(1).getStringValue();
+
+					XTable followUpTable = new XTable();
+					followUpTable.setId(dependencyTableName + transactionChecker);
+					Table innerTable = new Table();
+					innerTable.setName(transactionChecker);
+					innerTable.addColumn(new Column("KeyLong", DataType.INTEGER));
+					List<Row> innerTableRows = new ArrayList<>();
+
+					// Alle ResultSets mit diesem Namen (nicht ID) müssen gecheckt werden.
+					List<XSqlProcedureResult> resultsWithThatName = findxSqlResultSetByName(dependencyTableName, xsqlresults);
+
+					// Falls keine passenden OutputParameter gefunden werden können, muss das ResultSet des Haupt-Aufrufs (der erste in der Transaktion)
+					// verwendet werden.
+					if (resultsWithThatName.size() == 0) {
+						resultsWithThatName.add(xsqlresults.get(0));
+					}
+
+					// Und von diesen muss jede Row geprüft werden. Dabei holen wir uns jedes mal den KeyLong (siehe Doku).
+					for (XSqlProcedureResult res : resultsWithThatName) {
+						if (res.getResultSet().getOutputParameters() != null && res.getResultSet().getOutputParameters().getRows() != null) {
+							for (int i = 0; i < res.getResultSet().getOutputParameters().getRows().size(); i++) {
+								Value keyLongOfRow = findValueInColumn(res.getResultSet(), "KeyLong", i);
+
+								// Falls die Prozedur bzw. dessen ResultSet keinen KeyLong als Output hatte, greifen wir auf den KeyLong des Haupt-ResultsSets
+								// zurück.
+								if (keyLongOfRow == null) {
+									keyLongOfRow = findValueInColumn(xsqlresults.get(0).getResultSet(), "KeyLong", 0);
+								}
+								Row innerRow = new Row();
+								innerRow.addValue(keyLongOfRow);
+								innerTableRows.add(innerRow);
+								innerTable.setRows(innerTableRows);
+							}
+						}
+						followUpTable.setTable(innerTable);
+						xtables.add(followUpTable);
```

Comment:
Müsste man hier nicht jeweils eine nue followUpTable erzeugen,
bzw. die Erzeugung am Anfang dieser for-Schleife verschieben?